docs/generate-time-series/javascript.rst

```diff
@@ -0,0 +1,241 @@
+    message: 'success' }
+
+The endpoint returns a JSON payload consisting of ``iss_position``, which is composed of
+``latitude`` and ``longitude``, as well as some other metadata.
+
+You can encapsulate this within a single function to return only the longitude and latitude as a `WKT`_
+string::
+
+    > async function position() {
+    ...     let response = await axios.get('http://api.open-notify.org/iss-now.json')
+    ...     return `POINT (${response.data.iss_position.longitude} ${response.data.iss_position.latitude})`
+    ... }
+
+When you run this function, it should return your point string::
+
+    > await position()
+    'POINT (-22.7857 49.3332)'
+
+Set up CrateDB
+==============
+
+First, import the `pg`_ client::
+
+    > const { Client } = require('pg')
+
+Then `connect`_ to CrateDB, using the port for the `Postgres Wire Protocol`_::
+
+    > const client = new Client({connectionString: 'postgresql://crate@localhost:5432/doc'})
+    > await client.connect()
+
+Then finally `create a table`_ suitable for writing ISS position coordinates::
+
+    > var query = `
+    ...     CREATE TABLE iss_position (
+    ...         timestamp TIMESTAMP GENERATED ALWAYS AS CURRENT_TIMESTAMP,
+    ...         position GEO_POINT)`
+    > await client.query(query)
+
+
+In the `CrateDB Admin UI`_, you should see the new table when you navigate to
+the *Tables* screen using the left-hand navigation menu:
+
+.. image:: ../_assets/img/generate-time-series/table.png
+
+Record the ISS position
+=======================
+
+With the table in place, you can start recording the position of the ISS.
+
+The following command calls your ``position`` function and inserts the resulting
+position into your table using an `INSERT`_ query::
+
+    > await client.query("INSERT INTO iss_position (position) VALUES (?)", [await position()])
+
+Press the up arrow on your keyboard and hit *Enter* to run the same command a
+few more times.
+
+When you're done, you can `SELECT`_ that data back out of CrateDB, like so::
+
+    > let result = await client.query('SELECT * FROM iss_position')
+    > result.rows
+    [ { timestamp: 2020-02-21T09:44:12.754Z,
+        position: { x: -13.3941, y: 50.8954 } },
+    { timestamp: 2020-02-21T09:44:16.003Z,
+        position: { x: -13.0547, y: 50.9348 } },
+    { timestamp: 2020-02-21T09:44:17.035Z,
+        position: { x: -12.9576, y: 50.9458 } }]
+
+Here you have recorded three sets of ISS position coordinates.
+
+Automate it
+===========
+
+Now you have the basics figured out, let's automate the data collection. This will
+change your approach a little bit.
+
+You were previously using a `client`_ to connect to and insert data into CrateDB.
+However, clients are ephemeral, and once closed they need to be recreated.
+Creating a new client requires a handshake with CrateDB, and
+this overhead cost can be prohibitive if you are creating new clients rapidly.
+
+Instead, you can use a `connection pool`_ to manage your connections.
```

Comment:
perhaps add a brief sentence to this saying how connection pools address the problem